lib/modules/datasource/cpan/index.ts

```diff
@@ -0,0 +1,109 @@
+import type { HttpResponse } from '../../../util/http/types';
+import * as perlVersioning from '../../versioning/perl';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { MetaCpanFileSearchResult } from './types';
+
+export class CpanDatasource extends Datasource {
+  static readonly id = 'cpan';
+
+  constructor() {
+    super(CpanDatasource.id);
+  }
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultRegistryUrls = ['https://fastapi.metacpan.org/'];
+
+  override readonly defaultVersioning = perlVersioning.id;
+
+  @cache({
+    namespace: `datasource-${CpanDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `${packageName}`,
+  })
+  override async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    let result: ReleaseResult | null = null;
+    const searchUrl = `${registryUrl}v1/file/_search`;
+
+    let raw: HttpResponse<MetaCpanFileSearchResult> | null = null;
+    try {
+      const body = {
+        query: {
+          filtered: {
+            query: { match_all: {} },
+            filter: {
+              and: [
+                { term: { 'module.name': packageName } },
+                { exists: { field: 'module.associated_pod' } },
+              ],
+            },
+          },
+        },
+        _source: [
+          'module.name',
+          'module.version',
+          'distribution',
+          'date',
+          'deprecated',
+          'maturity',
+        ],
+        sort: [{ date: 'desc' }],
+      };
+      raw = await this.http.postJson<MetaCpanFileSearchResult>(searchUrl, {
+        body,
+      });
+    } catch (err) {
+      this.handleGenericErrors(err);
+    }
+
+    const body = raw?.body;
+    if (body) {
+      const hits = body.hits.hits;
+      const releases: (Release & { distribution: string })[] = [];
+      for (const hit of hits) {
+        const {
+          _source: {
+            module,
+            distribution,
+            date: releaseTimestamp,
+            deprecated: isDeprecated,
+            maturity,
+          },
+        } = hit;
+        const version = module.find(
+          ({ name }) => name === packageName
+        )?.version;
+        if (version) {
+          // https://metacpan.org/pod/CPAN::DistnameInfo#maturity
+          const isStable = maturity === 'released';
+          releases.push({
+            distribution,
+            // Release properties
+            isDeprecated,
+            isStable,
+            releaseTimestamp,
+            version,
+          });
+        }
+      }
+      if (releases.length > 0) {
+        const latest = releases[0];
+        result = {
+          releases,
+          changelogUrl: `https://metacpan.org/dist/${latest.distribution}/changes`,
```

Comment:
i don't think this makes sense, as renovate won't use non GitHub or gitlab urls.

@rarkins a I'm right?

Reply:
Right, but I think we might still embed a link in the PR, so it's still useful

Reply:
That url is linked here:

<img width="430" alt="image" src="https://user-images.githubusercontent.com/909674/180636519-554013e3-1f04-4e5d-988e-0ba5084b575e.png">

https://github.com/ikesyo/renovate-cpan-datasource-example/pull/9

Reply:
If this is not acceptable,  that means that non GitHub or gitlab urls in `manualChangelogUrls` should not be accepted as well.

Reply:
Let's leave it how you have it